lib/compat/wordpress-6.7/query-loop-block-filter.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Add date query to the query loop block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Filter the query loop block query vars.
+ *
+ * @param array  $query Query vars.
+ * @param object $block Block data.
+ *
+ * @return array
+ */
+function gutenberg_filter_query_loop_block_query_vars( $query, $block ) {
+
+	if ( ! empty( $block->context['query']['after'] ) || ! empty( $block->context['query']['before'] ) ) {
+		$query['date_query'] = array(
```

Comment:
We would also need server side validation for these params. Also, as @carolinan mentioned, they should be moved under `compat/wodpress-6.8`.